gcs/src/main/scala/monix/connect/gcp/storage/GcsBucket.scala

```diff
@@ -70,62 +74,69 @@ class GcsBucket private (underlying: Bucket) extends GcsDownloader with FileIO w
   }
 
   /**
-    * Allows downloading a Blob from GCS directly to the specified file.
+    * Downloads a Blob from GCS directly to the specified file.
     *
     * == Example ==
     *
-    * {
+    * {{{
     *   import java.io.File
     *
     *   import monix.connect.gcp.storage.{GcsStorage, GcsBucket}
     *   import monix.eval.Task
     *
-    *   val storage: GcsStorage = ???
-    *   val getBucketT: Task[Option[GcsBucket]] = storage.getBucket("myBucket")
+    *   val storage = GcsStorage.create()
     *   val targetFile = new File("example/target/file.txt")
     *   val t: Task[Unit] = {
     *     for {
-    *       maybeBucket <- getBucketT
+    *       maybeBucket <- storage.getBucket("myBucket")
     *       _ <- maybeBucket match {
     *         case Some(bucket) => bucket.downloadToFile("myBlob", targetFile.toPath)
     *         case None => Task.unit
     *       }
     *     } yield ()
     *    }
-    * }
+    * }}}
     */
+  @Unsafe("Risk of downloading large amounts of data into the local filesystem.")
   def downloadToFile(blobName: String, path: Path, chunkSize: Int = 4096): Task[Unit] = {
     val blobId = BlobId.of(underlying.getName, blobName)
     (for {
       bos   <- openFileOutputStream(path)
       bytes <- download(underlying.getStorage, blobId, chunkSize)
-    } yield bos.write(bytes)).completedL
+      _ <- try {
+        bos.write(bytes)
+        Observable.unit
+      } catch {
+        case NonFatal(ex) => Observable.raiseError(ex)
```

Comment:
Realised that this method was not safe since `bos.write(bytes)` can throw exceptions.
@Avasil do you like how it is handled now?

Reply:
Does it change anything? The error will be caught in `yield` before the change

Reply:
Mhh yeah you are right, it will be caught in the `Observable`